### ISSUE TITLE
Unfocusable tabs on /products and /orders pages 

### ISSUE THREAD

**Issue by Olaitan94** [Shopify/polaris-react-archive#9637]
## Issue summary

While testing the /products and /orders pages on the shopify admin for WCAG [1.1.1 Non-text Content (Level A)](https://www.w3.org/TR/WCAG20/#text-equiv-all) and [2.1.1 Keyboard (Level A)](https://www.w3.org/TR/WCAG20/#keyboard-operation-keyboard-operable), it was observed that some tabs on these pages are not focusable and are not identified by the screen reader. 

On the /products page, the following tabs are unfocusable: Active, Draft, and Archived.

On the /orders page, the following tabs are unfocusable: Unfulfilled, Unpaid, Open, and Closed.

This issue exists with ot without the polaris beta flag enabled

## Expected behavior

All functionality and buttons that are operable by clicking a mouse should also be operable using keyboard input only or a screen reader.

## Actual behavior

Certain tabs/buttons do not receive focus with keyboard input or screen reader. 

On the /products page, the following tabs are unfocusable: Active, Draft, and Archived.

On the /orders page, the following tabs are unfocusable: Unfulfilled, Unpaid, Open, and Closed.

## Steps to reproduce the problem

**Testing with Keyboard input** 
1. Go to Shopify admin
2. Navigate to /products page
3. Using the tab key on your keyboard, navigate through all the sections, tabs and buttons on the page. 
4. Identify the tabs that did not receive focus
5. Navigate to /orders page
6. Repeat steps 3 & 4

**Testing with Voice over** 
1. Go to Shopify admin
2. Navigate to /products page
3. Activate voice over with command and F5 . Navigate through all the sections, tabs and buttons on the page
4. Note the tabs that were not identified by voice over
5. Navigate to /orders page
6. Repeat steps 3 & 4

## Specifications

- Are you using the React components? (Y/N):
- Polaris version number: 
- Browser: Google chrome 
- Device: Apple M2 Pro
- Operating System: Max OS ventura 13.4

**Comment by chloerice** [comment]
Hey @Olaitan94 👋🏽 Tabs are not tabbable. The tabs are built to WCAG tab spec, where the first tab is in the tab order and the rest are navigated by arrow key. The reason the spec outlines this is because features should be tabbable without having to go through every single item in the list. So the tabs you listed for /products and /orders are accessible as expected, you need to tab to the tabs and then navigate with the arrow keys.

**Comment by mattkubej** [comment]
[Tabs specification](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) ([demo](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/examples/tabs-manual/))

**Comment by mattkubej** [comment]
Crosspost from Slack:

https://screenshot.click/07-33-602zq-x12rm.mp4

Here I’m using Mac’s VoiceOver for a screen reader. I Tab to the tabs, then I press Tab to navigate to the “Create view” icon, then press Tab+Shift to navigate back to the tabs. I navigate through the tabs with RightArrow and LeftArrow, then use Tab to navigate out of the tabs and back to the “Create view” icon. The screenreader seems to recognize the tabs and you’re able to navigate with the keyboard based on the w3 tab pattern we’re aligning with.
